connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -778,6 +778,62 @@ message ReleaseExecuteResponse {
   optional string operation_id = 2;
 }
 
+// GetErrorInfoRequest defines criteria for retrieving cached exceptions.
+message GetErrorInfoRequest {
+
+  // (Required)
+  //
+  // The session_id specifies a spark session for a user id (which is specified
+  // by user_context.user_id).
+  // The id should be an UUID string of the format `00112233-4455-6677-8899-aabbccddeeff`
+  string session_id = 1;
+
+  // User context
+  UserContext user_context = 2;
+
+  // (Required)
+  // Provide a list of error ids to get.
+  repeated string error_ids = 3;
+}
+
+// GetErrorInfoResponse returns list of exceptions matching the given error ids.
+message GetErrorInfoResponse {
+
+  message StackTraceElementInfo {
```

Comment:
Returning StackTraceElement in a structured way is good for Scala client to interpret. But may not be useful for clients implemented in other languages. I will add a field in the request to make returning structure `StackTraceElement` optional so that for Python Client, it can directly get the message with JVM stacktrace